bundles/admin/admin-hierarchical-layerlist/components/Group.js

```diff
@@ -0,0 +1,266 @@
+     *                                              }
+     *                                          }
+     * @return {Object}                         popup configuration:
+     *                                          {
+     *                                              popup: null,
+     *                                              buttons: null,
+     *                                              messages: null
+     *                                          }
+     */
+    getGroupAddingPopupConf: function(tool, id, parentId, opts) {
+        var me = this;
+        var returnObject = {
+            popup: null,
+            buttons: [],
+            message: null
+        };
+        var popup = Oskari.clazz.create('Oskari.userinterface.component.Popup');
+        var errorDialog = Oskari.clazz.create('Oskari.userinterface.component.Popup');
+        errorDialog.setId('admin-hierarchical-layerlist-group-error-popup');
+        returnObject.popup = popup;
+
+        // check at if editing group (id defined)
+        // then also add delete button
+        if (id) {
+            var btnDelete = Oskari.clazz.create('Oskari.userinterface.component.Button');
+            btnDelete.setTitle(me.locale('buttons.delete'));
+            btnDelete.addClass('delete');
+            btnDelete.setHandler(function() {
+                // check at group has no layers or subgroups
+                var group = me.layerService.getAllLayerGroups(id);
+                var isLayersOrGroups = (group && group.layers && group.layers.length > 0) ? true : false;
+                if (group.groups && group.groups.length > 0) {
+                    isLayersOrGroups = true;
+                }
+                if (isLayersOrGroups) {
+                    var dialog = Oskari.clazz.create('Oskari.userinterface.component.Popup');
+                    dialog.show(me.locale('errors.groupnameDeleteCheckLayers.title'), me.locale('errors.groupnameDeleteCheckLayers.message'));
+                    dialog.fadeout(5000);
+                } else {
+                    var data = {
+                        id: id
+                    };
+
+                    // ned be confirm deleteing ?
+                    //
+                    // Confirm dialog
+                    var confirmDialog = Oskari.clazz.create('Oskari.userinterface.component.Popup');
+                    var confirmBtnOk = Oskari.clazz.create('Oskari.userinterface.component.Button');
+                    var confirmBtnCancel = Oskari.clazz.create('Oskari.userinterface.component.Button');
+
+                    confirmBtnOk.addClass('primary');
+                    confirmBtnOk.setTitle(me.locale('buttons.ok'));
+                    confirmBtnOk.setHandler(function() {
+                        confirmDialog.close();
+                        me._deleteGroup(data, popup, opts.type, tool);
+                    });
+
+                    confirmBtnCancel.setTitle(me.locale('buttons.cancel'));
```

Comment:
If you would use OkButton and CancelButton (https://github.com/oskariorg/oskari-frontend/tree/master/bundles/framework/divmanazer/component/buttons) you wouldn't need to have the localizations in this bundle.